### PR TITLE
Rewrite code of conduct.

### LINKED ISSUE #10: Code of Conduct v2.0
Organizers can adopt a "Provisional v2.0" Code of Conduct by referencing this issue when announcing the code of conduct:

> Code of Conduct: https://github.com/cville/conduct
> Using "Provisional v2.0" which is outlined in #10 identifying inclusion and safety topics which the organizers are actively reviewing.

1. [x] Expand on inclusion before discussing harassment. The primary tool available to organizers is to act when there exists unwelcome behavior, but it's important to speak to the intent of "you belong here."
2. [x] Prioritize safety of attendees. This requires review in general. Specifically, expand on how to respond to assault.
3. [ ] Provide organizers with preparation. Provide a guide to talk through exercises before an event: what could happen and what to do about it. In addition to named risks, talk through real examples that exist in the community (and take caution when identifying people personally), how those were handled, and what was learned. Identify one or more other confirmed attendees as "deputies" before the event, confirm their understanding of the code of conduct, and ask them to join in taking action when called.
4. [x] Allow the person who reported an incident to decide what they would like to do personally (and give respectful space without adding to burden) before addressing an incident with an alleged offender.
5. [x] Empower organizers to act without needing a statement or intention from the harassed, by what they can observe and not based on an individual's experience or "how it feels."
6. [x] Act without warning in more cases.
7. [x] Clarify that "organizer" can be any member of the community who regularly runs events under the code of conduct, regardless of whether they are organizing _this_ event.
8. [x] Include unwelcome physical closeness.
9. [x] Specifically disallow repeated unwelcome invitations to participate in something/anything. Unwelcome means an individual has directly declined or opted not to reply to an invitation. A specific concern is when people invite others to non-professional events and do not take "no" for an answer, but this is open to the interpretation that repeat invitations to participate in anything at all could be experienced as a form of harassment, even if mild.
10. [x] Discourage open solicitation of input for someone to speak to the whole of a demographic or other status based on how they might personally identify.
11. [x] Make clear that targeting groups in a position of minority with any comment, particularly when the commenter does not identify as belonging to that position, could be open to perception that the comment  reinforces social structures of domination.
12. [x] Resolve as many [open issues](https://github.com/cville/conduct/issues) along the way.

Where possible, use language found in other codes of conduct, instead of creating it.

---

I've collected notes from the past year and am posting them here for other organizers to review, discuss, and use in provisional status. Why now? In a movement to advance our code of conduct, and not in direct response to any recent event.

In closing this issue, remember to tag versions and push tags to allow organizers to link to a specific version of the code of conduct.

### PR DESCRIPTION
**Pull request by rduplain**
Rewrite the code of conduct. Closes #10. In #10, I suggested:

> Where possible, use language found in other codes of conduct, instead of creating it.

I realized that this suggestion was in the way of progress. We have had much exposition on the code of conduct in the last month. This rewrite comes at the culmination of that work, namely of the Cville Slack admins (in order of admin start date: @rduplain, @andrew-boyd, @Meekohi, @fj, @caljess599).

For me personally, this rewrite has a full year of work behind it (as seen in #10). The text itself is the result of a combined several days of work from the #admin community of Cville Slack. Notwithstanding its origin, the code of conduct is intended for all groups and events, and being adaptable to non-Charlottesville communities is a first-class consideration.

You can contribute to this rewrite's review in any of the following ways:

* Share comments and suggestions in #admin on Cville Slack.
* Review using GitHub's [pull request review feature](https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/#submitting-your-review), or add to the comment thread below.
* Create a git branch from "v2.x-dev" and share any resulting commits.

Review the document in full:

* [GitHub (Official Format)](https://github.com/cville/conduct/tree/v2.x-dev#readme)

Alternative documents for review (which match content but not formatting):

* [Word .docx](https://www.dropbox.com/s/fqh3dahpxrjq18t/README.docx?dl=0)
* [PDF](https://www.dropbox.com/s/p0vwmbmgtlt0q2q/README.pdf?dl=0)

For suggestions of further rewrites, please make a clear proposal on an alternative approach to the code of conduct and suggest whether such a rewrite would be a "v3.x" outcome or should be considered prior to tagging v2.0.

The _Table of Contents_ is generated by [markdown-toc](https://github.com/jonschlinkert/markdown-toc). I'm currently using the output of this program manually in revisions. See the commit message at 018499232d19de6e0d373933c7918401d24115fa.

I plan to merge this to branch "v2.x" and tag v2.0 on Sat Sep 15 at 12noon US/Eastern. Please provide review by then, preferable with enough lead time for review. Of course, we will continue iterating this code of conduct beyond v2.0. If anything needs more time, simply flag a section as "this could use more discussion" and we can hold it out of the v2.0 release.

### CODE REVIEW

**Comment by rduplain** [comment]
Reminder to self: change GitHub's default branch from "v1.x" to "v2.x" upon merge.

**Comment by itruett** [comment]
Suggestion for rewriting the politics section to include other divisive topics:

### Discussing divisive topics in public spaces
 Unless a group or event identifies these subjects as on-topic, misconduct includes discussion or views of national politics, religion, and recreational drug use in public spaces of the community. Specific spaces may be identified as preferring these discussions, for which such discussion is acceptable. It is acceptable to advocate for creating spaces in the community in which these topics are appropriate.
 Notwithstanding, discussions about _policies_ are generally acceptable. There is a difference between discussing policies in the context of their impact on the community and sharing reactionary comments to recent headlines. While it is acceptable to share reactions generally, reactions to political headlines or controversial topics are only accepted when a group, event, or space identifies as political in nature.

- - -
I'm tempted to put firearms on the list, but maybe that's covered in politics? Maybe a comprehensive list of topics isn't the best idea?

**Comment by rduplain** [comment]
Copying Isaac's comment from Cville Slack:

> Politics got a whole section to itself. It's usual company includes religion and money. Maybe politics, religion, and recreational drug use belong in an opt-in category together?

**Comment by fj** [comment]
Nice work, @rduplain. I feel that you've managed the very delicate balance of representing concerns fairly while still making this a maximally inclusive space.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+*.db-journal
+*.doc
+*.docx
+.DS_Store
+/.emacs*
+__MACOSX
+*.out
+*.pdf
+*.pyc
+*.sw*
```
Comment (fj) [inline .gitignore]: I guess a `.gitignore` can't hurt, but any specific reason why all of these are here? There shouldn't be any executable code in this repo anyway.

File: README.md [hunk 2]
```diff
@@ -1,22 +1,277 @@
...
+
+* By supporting alternative narratives, shame no one for making a mistake.
+* Address malice swiftly and assertively.
+* Unpack issues privately where possible.
+* Speak publicly to make clear to others that the conduct is indeed _misconduct_ or is otherwise being evaluated as such.
+
+
+## Accessibility
+
+If you have a preference or need to support full participation as it relates to the Americans with Disabilities Act (ADA), or similar request, simply contact an organizer in advance. Events are sometimes hosted informally at an office or other venue which may not be fully equipped. With some lead time on a request, an organizer can seek to relocate an event to a venue which is appropriately laid out and equipped.
```
Comment (fj) [inline README.md]: Worth linking to the ADA here? It's at https://www.ada.gov/pubs/adastatute08.htm.

File: README.md [hunk 3]
```diff
@@ -1,22 +1,277 @@
...
+3. With fellow organizers, **discuss a plan** to resolve the issue of misconduct. Document details to support the plan.
+4. **Carry out that plan**.
+5. Follow up to **ensure that the air is clear** for the person affected, the person with whom you provided feedback on the code of conduct, and for the community.
+
+
+### When you need to ask someone to leave the community
+
+Following the previous section, find consensus with other organizers to remove a participant from the community, when a person:
+
+* compromises others' health and safety
```
Comment (fj) [inline README.md]: `/s/and/or/`, so that either is sufficient rather than both being necessary.

File: README.md [hunk 4]
```diff
@@ -1,22 +1,277 @@
...
+
+* A short version, which (a) is a quick read, and (b) has enough detail to understand the long version.
+* A long version, which makes very clear how to implement the code of conduct and respond to specific scenarios.
+
+The code of conduct is best served as a single document for visibility and transparency to everyone.
+
+
+### What about ...?
+
+The code of conduct is open source, and you are welcome to contact one of the authors directly (contact information is on GitHub profile pages), [open an issue](https://github.com/cville/conduct/issues/new), or send a [pull request](https://github.com/cville/conduct/pulls).
```
Comment (fj) [inline README.md]: I'd `/s/one of//`, since that makes it sound like you're only permitted to contact a maximum of one author.

**Review by fj — COMMENTED** [review]
I've added some feedback to address a few minor points.

File: .gitignore (hunk 1 above)
Comment (rduplain) [reply .gitignore]: I've been writing a temporary toc.md file, and I was considering using a tool such as [cog](https://nedbatchelder.com/code/cog/) to write the table of contents automatically in place. I also generated .docx and .pdf versions for review. Any time I add a .gitignore, I include the standard intermediate files.

Would you prefer to omit the .gitignore file?

**Comment by rduplain** [comment]
Thanks, @fj!

@itruett Thank you for raising an important point. Let's unpack (in the comment to follow).

I'll note that there's an open thread on whether the code of conduct speaks to the _practice_ of being inclusive, such that there is no such thing as perfection. Mistakes happen.

File: .gitignore (hunk 1 above)
Comment (fj) [reply .gitignore]: When I read a .gitignore, for the most part, my expectation is that either:

* the project _does_ generate those files as part of its construction or build pipeline, and they should be ignored; or
* the project _doesn't_ generate those files, but the files are common across many kinds of projects, but are also inconveniently generated by operating systems (e.g. `.DS_Store` from MacOS) or pervasive tooling (e.g. `.tmp` files from editors)

Neither seems to be true in the case of `.docx`, `.pdf`, and most of the others in the list. I don't feel strongly about not changing it, though.

Comment (rduplain) [reply .gitignore]: Good points. I'll remove the .gitignore since there's no automation in place currently. I added at a time when I was generating several of these files with tools outside the project.

**Comment by rduplain** [comment]
Unpacking [divisive topics](#issuecomment-419672345). I understand the concern to be "misconduct includes a wider range of divisive and opt-in topics than just politics."

Three categories beyond politics:

* religion
* recreational drug use
* potentially others

Of these, *religion* is specifically named in "social structures". We will not tolerate discrimination or dominant assumptions as it relates to religion. A section similar to, but separate from, the current politics section could describe how each person is entitled to religious and spiritual views, and how to create opt-in experiences.

I'm concluding that no such section or change is needed. If a comment or action assumes that everybody is of a given religion, or carries an implicit context that one spiritual view is the majority view, then it is already covered in the "social structure" section. In generating examples, I find that remaining religion-related comments fall squarely in the politics section as already written.

A counterexample would be very helpful if we don't have consensus on that conclusion. If posting the example would be insensitive, please contact me ([email is listed here](https://github.com/rduplain), and please only while this pull request is "open" in the event you are reading this at some point in the future).

Next, *recreational drug use.* Unless we're drinking coffee, which is generally acceptable, then I'm concluding that all topics fall under explicit (legalization politics vs policy) and implicit (the laws governing the jurisdiction of the group/event) conduct/misconduct.

That said, let me know if you would like to see drunkenness given particular attention in the code of conduct. Public intoxication already has relevant laws, and while this code of conduct covers groups/events in public spaces, we certainly don't want anyone to mistake other participants for "drinking buddies." We do cover adjacent topics, such as invitations to personal parties being misconduct.

A related concern is insensitivity toward alcoholism.

Finally, *potentially others* that we could enumerate. I suspect that each will fall into the pattern present here: adjacent issues are already covered, and stating a view on a divisive topic will quickly land in social structures or be at risk of heated politics. This last point, to me, indicates why politics stands on its own.

**Comment by itruett** [comment]
Thank you for taking the time to thoughtfully unpack this subject. I think you've covered it quite well.

In summary, as I understand it:

1. Discrimination on the basis of religious superiority/dominance is prohibited.
2. Discussion of philosophy of religion is permitted only in specifically designated opt-in spaces. e.g., if it would be considered on-topic for _Mi Yodeya_ (https://judaism.stackexchange.com/) it would be off-topic for general conversation covered by this Code of Conduct but accepted in designated spaces (subject to #1). This is because, for purposes of this Code of Conduct, such conversation is considered political.
3. Lawful recreational drugs are on-topic (e.g., how to brew coffee or tea (caffeine), or beer (alcohol)).
4. Illegal drugs as a topic are off-topic. This includes references evocative of illegal drug use (e.g., "420") or figures of speech ("he's smoking crack" in reference to someone behaving oddly) are prohibited as political commentary.

The overriding principle, of course, is to be kind and respectful and inclusive. Avoid topics that cause offense or make people uncomfortable.

There also seems to be an implicit imperative to comport oneself lawfully and not induce unlawful behavior in others. This probably doesn't need to be explicit, unless we want to specifically rule out something like unlawful civil disobedience or some other form of "justifiable" lawbreaking.

**Comment by rduplain** [comment]
Excellent redux, @itruett. I do think we can improve the code of conduct by adding sections on religion and "drugs" (in the broadest sense, including coffee) which match the discussion of intent here.

**Comment by rduplain** [comment]
In #admin on Cville Slack, we have a discussion flagging "violating digital security". Specifically, in my interpretation:

* It's too broad, as "risk" can be interpreted incorrectly, especially by people who don't dedicate time to network security topics.
* The ability to fork the code of conduct could make this broad interpretation cause major issues in other communities.

There are likely to be more comments on it. The request currently is to clarify intent.

* The short version is "don't honeypot others."
* Do not target other members of the community with "cyber" attacks.
* If you are running an experiment, such as setting up a compromised wifi hotspot, do so only with opt-in from other participants. Do not have others participate unknowingly.

It's important to improve this section accordingly.

**Comment by rduplain** [comment]
We also had a reaction to the one-week review period. I set the period, so I'll clarify my intent:

* We can iterate forever.
* I've been part of open source (and company) projects where pull requests of this magnitude never get closed because there's always more.
* I hope that it's clear that we'll take every comment seriously, as to remove sections which have no consensus.

I'll underscore this in the pull request description above:

> For suggestions of further rewrites, please make a clear proposal on an alternative approach to the code of conduct and suggest whether such a rewrite would be a "v3.x" outcome or should be considered prior to tagging v2.0.

As part of the one-week review, it's an invitation to catch big issues and "pull the cord."

**Comment by rduplain** [comment]
I added this to the pull request description above. Underscoring here:

> If anything needs more time, simply flag a section as "this could use more discussion" and we can hold it out of the v2.0 release.

File: README.md [hunk 5]
```diff
@@ -1,22 +1,277 @@
...
+
+Any conduct which compromises health or safety of any other person, or otherwise increases such risks, will be escalated to emergency services.
+
+
+### Violating personal space
+
+Misconduct includes deliberate intimidation, unwelcome gestures, unwelcome physical closeness, inappropriate physical contact, stalking, following, unwelcome photography or recording, and unwelcome attention.
+
+
+### Violating digital security
```
Comment (Meekohi) [inline README.md]: Flagging this as "potentially controversial" for 2.x

File: README.md [hunk 6]
```diff
@@ -1,22 +1,277 @@
...
+
+Misconduct includes any behavior which compromises a computer's security, availability of services, personal online privacy, or privacy and security of digital documents. Misconduct further includes encouraging others to participate in an activity which knowingly increases risk of privacy and security issues.
+
+
+### Reinforcing social structures of domination
+
+Misconduct includes any comment or action of discrimination or which reinforces social structures of domination, as they relate to color, ancestry, national origin, immigration or naturalization status, actual or presumed belonging to an ethnic group, veteran status, religion, age, sex, sexual orientation, gender identity or expression, marital or family status, physical appearance, physical or mental disability, medical condition, education, occupation, income, or socioeconomic status.
```
Comment (Meekohi) [inline README.md]: convoluted wording... typo, I think? 
Misconduct includes any comment or action of discrimination _or_ which reinforces social structures

**Comment by rduplain** [comment]
Remaining topics:

* I held back the digital security section (c622d8150f3b3f8c4cc105ed17a45ffebc4dfcc6).
* We have consensus on opt-in topics (religion, drug use) above, but these could be improved (name opt-in nature of religion specifically, show sensitivity alcoholism, ...).

Iteration can address these.
